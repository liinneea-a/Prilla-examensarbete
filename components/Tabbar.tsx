import { Feather } from "@expo/vector-icons";
import React from "react";
import { View, Text, StyleSheet, Image } from "react-native";

const Tabbar = () => {
  return (
    <View style={styles.container}>
      <View style={styles.tabb}>
        <View style={styles.wrapper}>
          <View style={styles.logoWrapper}>
            <Image
              style={styles.logo}
              source={require("../assets/images/trending.png")}
            />
            <Feather
              style={styles.icon}
              name="trending-up"
              size={30}
              color="white"
            />
          </View>
          <Text style={styles.text}>Trendande sorter</Text>
        </View>
      </View>

      <View style={styles.tabb}>
        <View style={styles.wrapper}>
          <View style={styles.logoWrapper}>
            <Image
              style={styles.logo}
              source={require("../assets/images/toppbetyg.png")}
            />
            <Feather
              style={styles.icon}
              name="bar-chart-2"
              size={34}
              color="white"
            />
          </View>

          <Text style={styles.text}>Toppbetyg</Text>
        </View>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    justifyContent: "space-between",
    width: "100%",
    height: 70,
    marginTop: 10,
  },
  wrapper: {
    flexDirection: "row",
    alignItems: "center",
  },
  tabb: {
    justifyContent: "space-around",
    width: "100%",
    height: 50,
    backgroundColor: "#3D3745",
    marginTop: 10,
    marginLeft: 10,
    borderTopLeftRadius: 6,
    borderBottomLeftRadius: 6,
  },
  text: {
    color: "white",
    fontSize: 20,
    marginLeft: 20,
  },
  logo: {
    borderTopLeftRadius: 6,
    borderBottomLeftRadius: 6,
    height: 50,
    width: 50,
    opacity: 0.5,
  },
  logoWrapper: {
    position: "relative",
  },
  icon: {
    position: "absolute",
    top: 10,
    right: 10,
  },
});
export default Tabbar;
