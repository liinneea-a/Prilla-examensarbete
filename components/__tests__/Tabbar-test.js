import * as React from "react";
import renderer from "react-test-renderer";
import Tabbar from "../Tabbar";

it("should contain two text tags", () => {
  const tabb = renderer.create(<Tabbar />);
  const instance = tabb.root;
  const trending = instance.findByProps({ testID: "trending" });
  const popular = tabb.root.findByProps({ testID: "popular" });
  expect(trending.props.children).toBe("Trendande sorter");
  expect(popular.props.children).toBe("Toppbetyg");
  expect(tabb).toMatchSnapshot();
});
