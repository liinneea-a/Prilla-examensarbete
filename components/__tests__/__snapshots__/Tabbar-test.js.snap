// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should contain two text tags 1`] = `
<View
  style={
    Object {
      "justifyContent": "space-between",
      "marginBottom": 10,
      "marginTop": 10,
      "width": "100%",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "borderBottomLeftRadius": 6,
          "borderTopLeftRadius": 6,
          "height": 50,
          "justifyContent": "space-around",
          "marginLeft": 10,
          "marginTop": 10,
          "width": "100%",
        },
        Object {
          "backgroundColor": "#C6ADE8",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "position": "relative",
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "borderBottomLeftRadius": 6,
              "borderTopLeftRadius": 6,
              "height": 50,
              "opacity": 0.5,
              "width": 50,
            }
          }
        />
        <Text />
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "white",
            },
            Object {
              "color": "white",
              "fontSize": 20,
              "marginLeft": 20,
            },
          ]
        }
        testID="trending"
      >
        Trendande sorter
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "borderBottomLeftRadius": 6,
          "borderTopLeftRadius": 6,
          "height": 50,
          "justifyContent": "space-around",
          "marginLeft": 10,
          "marginTop": 10,
          "width": "100%",
        },
        Object {
          "backgroundColor": "#C6ADE8",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "position": "relative",
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "borderBottomLeftRadius": 6,
              "borderTopLeftRadius": 6,
              "height": 50,
              "opacity": 0.5,
              "width": 50,
            }
          }
        />
        <Text />
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "white",
            },
            Object {
              "color": "white",
              "fontSize": 20,
              "marginLeft": 20,
            },
          ]
        }
        testID="popular"
      >
        Toppbetyg
      </Text>
    </View>
  </View>
</View>
`;
